Switch to TypeScript

popa-marius opened the pull request:
Implements the planned switch to TypeScript from the [Roadmap card](https://github.com/codash-platform/codash/projects/1#card-36488416).

halvardssm committed:
added typescript

halvardssm committed:
rewrote files to tsx

halvardssm committed:
Merge pull request #1 from codash-platform/master

Update to match remote

halvardssm committed:
Merge branches 'feat/ts' and 'master' of github.com:halvardssm/codash into feat/ts

 Conflicts:
	src/global/sagas.tsx
	src/pages/dashboard/graphs/Graphs.tsx

halvardssm committed:
prettied

halvardssm committed:
changed eslint parser to typescript

popa-marius committed:
Merge branch 'master' of github.com:codash-platform/codash into feat/ts

 Conflicts:
	src/global/i18n.js

popa-marius committed:
Merge pull request #1 from halvardssm/feat/ts

Switch to typescript (partial change)

popa-marius committed:
Add missing types to make application compilable

popa-marius commented:
@halvardssm This commit has the changes needed in order to make the app compile.
https://github.com/codash-platform/codash/pull/2/commits/b98046500baea63dc112cf7d4f54681e8a70be8b

halvardssm commented:
Will have a go at finishing this PR this week 👍

popa-marius committed:
Merge branches 'master' and 'typescript' of github.com:codash-platform/codash into typescript

 Conflicts:
	package-lock.json
	package.json
	src/global/i18n.js

popa-marius committed:
Update most dependencies to latest versions

popa-marius committed:
Prettify code

popa-marius committed:
Fix hot reloading typescript migration

popa-marius committed:
Merge branch 'master' of github.com:codash-platform/codash into typescript

popa-marius committed:
Separate ts from tsx files for clearer file contents

halvardssm committed:
WIP: specifying types

halvardssm commented:
@popa-marius I am trying to push the changes I made to the types, but I dont have push access to this repo, could you change my access to "write"?

halvardssm committed:
WIP: Improving types

halvardssm committed:
prettied

halvardssm committed:
Fixed startup issue

popa-marius commented:
> @popa-marius I am trying to push the changes I made to the types, but I dont have push access to this repo, could you change my access to "write"?

Done. Please retry and let me know if there are any more problems. I'll also change my notifications, so I will be updated faster about comments.

halvardssm commented:
@popa-marius Works! It is ready now, so let me know what you think and if you want any changes 💪

popa-marius commented:
I haven't forgotten about this PR. I did a lot of travelling recently and had little time to attend to it, unfortunately. But I will focus on it in the next days 😃

halvardssm commented:
No worries! I have been busy with uni and work, and a vacation is never unfortunate 😉  Have fun!

popa-marius committed:
Merge branches 'typescript', 'master' and 'typescript' of github.com:codash-platform/codash into typescript

 Conflicts:
	package-lock.json
	package.json

popa-marius committed:
Update with latest master

popa-marius committed:
Fix ECDC data format for population with fallback

popa-marius committed:
Switch to null coalescing default values

popa-marius committed:
Further improve type handling

popa-marius committed:
Fix eslint warnings and errors

popa-marius commented:
@halvardssm Good work! I can only imagine the time it took to change so much of the code.
I continued on your work and did some additional improvements and cleaned some of the code. You can see it here https://github.com/codash-platform/codash/pull/2/files/4595333653e5a3cb98a6ac7ee0b75808eb4f795b..3a0d3868b46d197e34a2379f320e925998548359
On top of that, I updated it with the latest master and dependencies.

Please let me know if you want to look over it, otherwise I think we can merge it.

halvardssm commented on src/pages/dashboard/tables/Tables.tsx:
```diff
@@ -7,11 +7,11 @@ import {getTableData} from '../../../global/dataParsing'
 interface TableComponentProps extends WithTranslation {
   overview: Overview;
-  tableOverview: TableOverview;
+  tableOverview: {};
```
There is no specific type for the tableOverview?

popa-marius replied:
There was, but it didn't play nicely with the `TABLE_TYPE` because it was a const instead of an enum. I could not think of a better structure at the time.
![Screenshot 2020-07-14 at 10 09 03](https://user-images.githubusercontent.com/3997576/87401673-b9cc8a00-c5ba-11ea-8d33-b4614b587bae.png)

After looking a bit more over it, I think I came up with a somewhat clean solution by creating another enum, so I did a commit now.

halvardssm replied:
Nice 👍

halvardssm reviewed (approved):
Good thing you had a look over it as well 💪 Looks good to me

popa-marius committed:
Improve utility types consistency

popa-marius committed:
Improve Readme contributors description
